app/controllers/application_controller.rb

```diff
@@ -2,4 +2,40 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user, :logged_in?
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def correct_user
+    id = params[:user_id] || params[:id]
+    @user = User.find(id)
+    if @user == current_user
+    else
```

Comment:
The empty `if` clause here is a bit unusual. It'd probably read better if we said `if @user != current_user` and removed the `else` line.